File: app/src/main/java/star/iota/acgrip/ui/fan/FanContract.java
package star.iota.acgrip.ui.fan;

import java.util.List;

interface FanContract {

    interface View {

        void success(List<FansBean> fans);

        void error();

        void noData();
    }

    interface Presenter {
        void request(String url);

        void cancel();
    }
}

File: app/src/main/java/star/iota/acgrip/ui/fan/FanContract.kt
package star.iota.acgrip.ui.fan

import star.iota.jptv.base.BasePresenter
import star.iota.jptv.base.BaseView

internal interface FanContract {
    interface View : BaseView {
        fun success(fans: List<FansBean>)
    }

    interface Presenter : BasePresenter {
        fun request(url: String)
    }
}
